calblueprint/dccentralkitchen-clerks
# Firebase Analytics + Sentry error logging

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
* Integrated Firebase Analytics (new compatibility with Expo SDK 37 from #52 ) as a free alternative to FullStory. Getting a clear view of how clerks navigate the app can hopefully assist with remote user testing and expose bugs or points of confusion in the UX.
* Added Analytics logging for the following events:
  - ClerkLogOut
  - GoBackApplyRewards
  - ConfirmTransaction
  - ClerkLogin
  - CustomerFound
  - SelectTrainingMode
  - ConfirmQuantity
  - OpenQuantityModal
  - OpenRewardsModal
  - ConfirmApplyRewards
* Added Sentry error logging in all try/catch blocks

## Relevant Links

### Online sources
[Firebase setup](https://docs.expo.io/guides/setup-native-firebase/?redirected)
[Expo Firebase Analytics](https://docs.expo.io/versions/latest/sdk/firebase-analytics/)

### Related PRs
Analogous to https://github.com/calblueprint/dccentralkitchen/pull/88/

## How to review
* ~~In `App.js` set `  Analytics.setAnalyticsCollectionEnabled(false)` to `true`.~~
* Log into the [Firebase console](https://console.firebase.google.com/) (shared with healthyrewards@calblueprint.org , credentials pinned in Slack)
* In the console, go to `StreamView` under `Analytics` in the menu on the left
* Click around the app -- complete test transactions, clerk login, clerk training mode etc and see if the events are properly logged. You may need to refresh the Firebase console a few times for events to start showing up.
* Look through the Events and various parameters and make sure everything is accurate. Look out for anything that's marked `undefined`.
* Make sure the event names and parameters all make sense

## Next steps
* Possibly clean up the event logging into functions in `logUtils` like Sentry. This might be difficult to standardize since the parameters attached to each event vary so much, but 
* More extensive event logging--look into tracking each transaction process as a session instead of individual events

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases
* Tested several transactions/combinations etc.

### Screenshots
* Successful event logging in the Firebase Console
![analytics events](https://user-images.githubusercontent.com/21160510/80273424-bd6d0680-8686-11ea-81ea-d2098108e211.png)


CC: @anniero98 

[//]: # 'This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
